charts/karpenter/templates/deployment.yaml

```diff
@@ -113,6 +113,12 @@ spec:
           volumeMounts:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+        {{- with .Values.controller.sidecarContainer }}
+          {{- with .Values.controller.extraVolumeMounts }}
+          volumeMounts:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+        {{- end }}
```

Comment:
How is this supposed to be used? `sidecarContainer` value will never be injected here, this seems like some copy-paste gone wrong here
```
❯ helm template . --set controller.sidecarContainer.name="hello world" --debug

Error: template: karpenter/templates/deployment.yaml:117:26: executing "karpenter/templates/deployment.yaml" at <.Values.controller.extraVolumeMounts>: nil pointer evaluating interface {}.controller
helm.go:84: [debug] template: karpenter/templates/deployment.yaml:117:26: executing "karpenter/templates/deployment.yaml" at <.Values.controller.extraVolumeMounts>: nil pointer evaluating interface {}.controller
```